packages/editor/src/lib/components/default-components/DefaultShapeIndicators.tsx

```diff
@@ -28,27 +29,31 @@ export const DefaultShapeIndicators = memo(function DefaultShapeIndicators({
 		'should display selected ids',
 		() => {
 			const prev = rPreviousSelectedShapeIds.current
-			const next = new Set<TLShapeId>()
+			if (hideAll) {
+				return prev
+			}
 
-			const instanceState = editor.getInstanceState()
+			let shouldShowIndicators = false
+			let state: StateNode | undefined = editor.getCurrentTool()
+			// allow deeper state nodes to override the should show indicators value of shallower nodes
+			while (state) {
+				shouldShowIndicators = !!state.getShouldShowIndicators()
+				state = state.getCurrent()
+				if (!state) break
+			}
```

Comment:
i'd be tempted to make this a method on `editor` or the root state node for easier reuse. wdyt?